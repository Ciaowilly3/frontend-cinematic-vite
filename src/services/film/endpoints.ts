import {
  BaseQueryFn,
  EndpointBuilder,
  FetchArgs,
  FetchBaseQueryError,
  FetchBaseQueryMeta,
} from "@reduxjs/toolkit/query";
import { HTTP } from "../../enums/HttpMethodsEnum";
import { Routes } from "../../enums/routesEnums";
import { IFilm, IFilmDto, IFilms } from "../../interfaces/IFilm";

type customBuilder = EndpointBuilder<
  BaseQueryFn<
    string | FetchArgs,
    unknown,
    FetchBaseQueryError,
    {},
    FetchBaseQueryMeta
  >,
  "films",
  "filmsApi"
>;

const retrieveAllFilms = (builder: customBuilder) =>
  builder.query<IFilms, void>({
    query: () => Routes.films,
    providesTags: ["films"],
  });
const makeNewFilm = (builder: customBuilder) =>
  builder.mutation<IFilm, IFilmDto>({
    query: (body) => ({
      url: Routes.films,
      method: HTTP.POST,
      body,
    }),
    invalidatesTags: ["films"],
  });
const updateFilmById = (builder: customBuilder) =>
  builder.mutation<IFilm, { id: string; body: IFilmDto }>({
    query: ({ id, body }) => ({
      url: `${Routes.films}/private/update-film/${id}`,
      method: HTTP.PUT,
      body: body,
    }),
    invalidatesTags: ["films"],
  });
const deleteFilmById = (builder: customBuilder) =>
  builder.mutation<void, string>({
    query: (id) => ({
      url: `${Routes.films}/delete-film/${id}`,
      method: HTTP.DELETE,
    }),
    invalidatesTags: ["films"],
  });
export { retrieveAllFilms, makeNewFilm, updateFilmById, deleteFilmById };
