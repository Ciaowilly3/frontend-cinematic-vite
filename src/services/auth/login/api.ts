import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { BASE_URL } from "../../utils";
import { login, register } from "./endpoints";

export const loginApi = createApi({
  reducerPath: "loginApi",
  baseQuery: fetchBaseQuery({ baseUrl: BASE_URL + "auth/" }),
  endpoints: () => ({}),
}).injectEndpoints({
  endpoints: (builder) => ({
    login: login(builder),
    register: register(builder),
  }),
});

export const { useLoginMutation, useRegisterMutation } = loginApi;

//TODO: salvare token nei cookies
//TODO: provare a mantenere la sessione valida al ricaricare
