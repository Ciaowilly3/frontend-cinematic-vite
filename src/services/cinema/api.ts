import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
import { retrieveAllCinema } from './endpoints'

export const BASE_URL : string = 'http://localhost:8080/'

export const cinemaApi = createApi({
    reducerPath: 'cinemaApi',
    baseQuery: fetchBaseQuery({baseUrl: BASE_URL}),
    endpoints: () => ({})
}).injectEndpoints({
    endpoints: (builder) => ({
        retrieveAllCinema : retrieveAllCinema(builder)
    })
})

export const { useRetrieveAllCinemaQuery } = cinemaApi