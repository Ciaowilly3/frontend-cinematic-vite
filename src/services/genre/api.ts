import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { BASE_URL } from "../utils";
import { authTokenState } from "../../slices/auth/authTokenSlice";
import { makeNewGenre, retrieveAllGenres } from "./endpoints";
import { RootState } from "../../store/store";

export const genresApi = createApi({
  tagTypes: ["genres"],
  reducerPath: "genresApi",
  baseQuery: fetchBaseQuery({
    baseUrl: BASE_URL,
    prepareHeaders: (headers, { getState }) => {
      const authToken: authTokenState = (getState() as RootState).authToken;

      authToken.token
        ? headers.set("authorization", `Bearer ${authToken.token}`)
        : "";
      return headers;
    },
  }),
  endpoints: () => ({}),
}).injectEndpoints({
  endpoints: (builder) => ({
    retrieveAllGenres: retrieveAllGenres(builder),
    makeNewGenre: makeNewGenre(builder),
  }),
});

export const { useRetrieveAllGenresQuery, useMakeNewGenreMutation } = genresApi;
