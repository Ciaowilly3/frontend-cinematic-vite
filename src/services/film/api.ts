import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
import { BASE_URL } from "../cinema/api";
import { makeNewFilm, retrieveAllFilms } from "./endpoints";

export const filmApi = createApi({
  reducerPath: "filmsApi",
  baseQuery: fetchBaseQuery({ baseUrl: BASE_URL }),
  endpoints: () => ({}),
}).injectEndpoints({
  endpoints: (builder) => ({
    retrieveAllFilms: retrieveAllFilms(builder),
    makeNewFilm: makeNewFilm(builder),
  }),
});

export const { useRetrieveAllFilmsQuery, useMakeNewFilmMutation } = filmApi;
