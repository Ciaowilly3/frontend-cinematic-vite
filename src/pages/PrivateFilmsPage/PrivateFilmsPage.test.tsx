import { render, waitFor } from '@testing-library/react';
import PrivateFilmPage from './PrivateFilmsPage';
import { useRetrieveAllFilmsQuery } from '../../services/film/api';

jest.mock('../../components/GenreInput');
jest.mock('../../services/film/api');

const mockedUseRetrieveAllFilmsQuery = jest.mocked(useRetrieveAllFilmsQuery);

const renderComponent = () => render(<PrivateFilmPage />);

describe('PrivateFilmPage', () => {
  test('should render', () => {
    mockedUseRetrieveAllFilmsQuery.mockReturnValue({} as never);
    const view = renderComponent();

    expect(view).toMatchSnapshot();
  });
  test('should display error when query fails', async () => {
    mockedUseRetrieveAllFilmsQuery.mockReturnValue({
      isError: true,
    } as never);
    const { getByText } = renderComponent();

    await waitFor(() => {
      expect(getByText('ERROR')).toBeInTheDocument();
    });
  });
});
