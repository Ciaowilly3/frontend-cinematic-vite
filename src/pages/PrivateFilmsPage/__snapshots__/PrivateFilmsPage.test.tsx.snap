// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivateFilmPage should render 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <h3
          class="text-my-primary"
        >
          PrivateFilmPage
        </h3>
        <div>
          <div
            class="my-3 d-flex justify-content-between align-items-center"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
                />
              </svg>
              Create a film
            </button>
          </div>
          <div />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h3
        class="text-my-primary"
      >
        PrivateFilmPage
      </h3>
      <div>
        <div
          class="my-3 d-flex justify-content-between align-items-center"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
              />
            </svg>
            Create a film
          </button>
        </div>
        <div />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
